java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -410,6 +411,31 @@ private void visit(Tree tree, Tree terminator) {
     if (tree.parent().is(Tree.Kind.EXPRESSION_STATEMENT)) {
       programState = ProgramState.unstack(programState, programState.stack.size()).a;
     }
+    programState = removeUnreachableSymbolicValues(programState);
+  }
+
+  private ProgramState removeUnreachableSymbolicValues(final ProgramState ps) {
+    final List<SymbolicValue> unreachableValues = new ArrayList<>();
+    ps.constraints.forEach(new PMap.Consumer<SymbolicValue, Object>() {
+      @Override
+      public void accept(SymbolicValue sv, Object value) {
+        if (sv!= SymbolicValue.NULL_LITERAL && sv!=SymbolicValue.TRUE_LITERAL && sv!=SymbolicValue.FALSE_LITERAL &&
+            !(sv instanceof SymbolicValue.BinarySymbolicValue || sv instanceof SymbolicValue.UnarySymbolicValue) && !ps.isReachable(sv) && !ps.stack.contains(sv)) {
+          unreachableValues.add(sv);
+        }
+      }
+    });
+
+    PMap<SymbolicValue, Object> cleanedConstraints = ps.constraints;
+    for (SymbolicValue unreachableValue : unreachableValues) {
+      cleanedConstraints = cleanedConstraints.remove(unreachableValue);
+    }
+
+    if(ps.constraints != cleanedConstraints) {
+      ProgramState programState = new ProgramState(ps.values, cleanedConstraints, ps.visitedPoints, ps.stack);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "programState" which hides the field declared at line 86. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "programState". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1488)